src/bentoml/_internal/marshal/dispatcher.py

```diff
@@ -217,6 +211,10 @@ async def train_optimizer(
                 if batch_size > 1:  # only wait if batch_size
                     a = self.optimizer.o_a
                     b = self.optimizer.o_b
+                    wait = min(
+                        self.max_latency * 0.95,
+                        (batch_size * 2 + 1) * (a + b),
+                    )
```

Comment:
I don't think moving wait here solves the issue

Since wait should already be defined as before

Reply:
It does since `wait` is only used by L219 below. But it seems to change the behavior since `a`, `b` and `self.max_latency` all change during the loop while `wait` doesn't change before.

Reply:
I agree this fix does not good solution . it's just temporary fix

It is a temporary fix Pull Request aimed at hotfix.

currently we can not use `workers_per_resource Option` in BentoML `v1.1.0 ~v1.1.1`

Reply:
@sauyon does this look ok with you?